packages/@lwc/engine-core/src/framework/check-version-mismatch.ts

```diff
@@ -14,8 +14,8 @@ import { LightningElementConstructor } from './base-lightning-element';
 
 let warned = false;
 
-// @ts-ignore
-if (process.env.NODE_ENV !== 'production' && typeof __karma__ !== 'undefined') {
+// Only used in LWC's Karma tests
+if (process.env.NODE_ENV === 'test-karma-lwc') {
```

Comment:
The reason I reuse `process.env.NODE_ENV` in this weird way is because we still need to ship `process.env.NODE_ENV` to consumers. And I couldn't invent some new variable (e.g. `process.env.IS_KARMA`) because that would break anyone who isn't replacing that string.

`process.env.NODE_ENV === 'test-karma-lwc'` seemed like the least-bad option.

Reply:
What was the reasoning behind needing a new variable or new value for `NODE_ENV`? (vs just checking `__karma__`?) Is it mainly to help with the transformations?

Reply:
If we were to just use `typeof __karma__ !== 'undefined'` then the code would not be tree-shaken out in production environments.

Reply:
Unfortunately `NODE_ENV === 'test'` was already [squatted](https://jestjs.io/docs/environment-variables#node_env) by Jest. And [we have](https://github.com/search?q=repo%3Asalesforce%2Flwc%20node_env%20%3D%3D%3D%20%27test%27&type=code) some `NODE_ENV === 'test'` conditions in our code that are explicitly designed for Jest (and _not_ for Karma). 😞 

I felt that `test-karma-lwc` made it super-obvious what this condition should be used for.